src/virtualenv/util/lock.py

```diff
@@ -72,11 +73,17 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self._release(self._lock)
 
     def _lock_file(self, lock):
-        ensure_dir(self.path)
+        # multiple processes might be trying to get a first lock... so we cannot check if this directory exist without
+        # a lock, but that lock might then become expensive, and it's not clear where that lock should live.
+        # Instead here we just ignore if we fail to create the directory.
+        try:
+            os.makedirs(str(self.path))
+        except OSError:
+            pass
```

Comment:
this should only catch the exception if it's EEXIST and it's a directory I believe -- other OSErrors should raise here (in py3+ `os.makedirs(..., exist_ok=True)`)

Reply:
not too worried about this, as if this fails for any other reason than EEXIST the next line will also fail, and done it like this to handle both platforms 🤔 I know about exists_ok=True... but that does not work on python 2...